src/vm/hints/keccak_utils.rs

```diff
@@ -109,9 +109,130 @@ pub fn unsafe_keccak(
     }
 }
 
+/*
+Implements hint:
+
+    %{
+        from eth_hash.auto import keccak
+        keccak_input = bytearray()
+        n_elms = ids.keccak_state.end_ptr - ids.keccak_state.start_ptr
+        for word in memory.get_range(ids.keccak_state.start_ptr, n_elms):
+            keccak_input += word.to_bytes(16, 'big')
+        hashed = keccak(keccak_input)
+        ids.high = int.from_bytes(hashed[:16], 'big')
+        ids.low = int.from_bytes(hashed[16:32], 'big')
+    %}
+
+ */
+pub fn unsafe_keccak_finalize(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    /* -----------------------------
+    Just for reference (cairo code):
+    struct KeccakState:
+        member start_ptr : felt*
+        member end_ptr : felt*
+    end
+    ----------------------------- */
+
+    let keccak_state_ptr =
+        match get_address_from_var_name("keccak_state", &ids, vm, hint_ap_tracking) {
+            Ok(MaybeRelocatable::RelocatableValue(relocatable)) => relocatable,
+            Err(e) => return Err(e),
+            _ => unreachable!(),
+        };
+
+    // as `keccak_state` is a struct, the pointer to the struct is the same as the pointer to the first element.
+    // this is why to get the pointer stored in the field `start_ptr` it is enough to pass the variable name as
+    // `keccak_state`, which is the one that appears in the reference manager of the compiled JSON.
+    let start_ptr = get_ptr_from_var_name("keccak_state", &ids, vm, hint_ap_tracking)?;
+
+    // in the KeccakState struct, the field `end_ptr` is the second one, so this variable should be get from
+    // the memory cell contiguous to the one where KeccakState is pointing to.
+    let end_ptr = vm.memory.get_relocatable(&Relocatable {
+        segment_index: keccak_state_ptr.segment_index,
+        offset: keccak_state_ptr.offset + 1,
+    })?;
+
+    // this is not very nice code, we should consider adding the sub() method for Relocatable's
+    let maybe_rel_start_ptr = MaybeRelocatable::RelocatableValue(start_ptr.clone());
+    let maybe_rel_end_ptr = MaybeRelocatable::RelocatableValue(end_ptr.clone());
+
+    let n_elems = match maybe_rel_end_ptr.sub(&maybe_rel_start_ptr, &vm.prime) {
+        Ok(MaybeRelocatable::Int(num)) => num
+            .to_usize()
+            .ok_or(VirtualMachineError::BigintToUsizeFail)?,
+        Err(e) => return Err(e),
+        _ => unreachable!(),
+    };
+
+    let mut keccak_input = Vec::new();
+    let range = vm
+        .memory
+        .get_range(&maybe_rel_start_ptr, n_elems)
+        .map_err(VirtualMachineError::MemoryError)?;
+
+    assert_no_nones_in_range(&range)?;
```

Comment:
Calling it "assert" may lead to confusion.
```suggestion
    check_no_nones_in_range(&range)?;
```